pyquil/api/_engagement_manager.py

```diff
@@ -23,13 +23,26 @@
 from qcs_api_client.models import EngagementWithCredentials, CreateEngagementRequest
 from qcs_api_client.operations.sync import create_engagement
 from qcs_api_client.types import UNSET
+from qcs_api_client.util.errors import QCSHTTPStatusError
 
 from pyquil.api._qcs_client import qcs_client
 
 if TYPE_CHECKING:
     import httpx
 
 
+class QPUUnavailableError(Exception):
+    """
+    Exception raised when a QPU is unavailable due to maintenance.
+    """
+
+    retry_after: Optional[int]
+    """The number of seconds after which to retry the engagement request."""
+
+    def __init__(self, retry_after: Optional[str]) -> None:
+        super().__init__(f"QPU unavailable due to maintenance. Please retry after {retry_after}s.")
```

Comment:
If `retry_after` is `None`, won't this message read a bit strangely?

Reply:
Can't handle some strangeness, Graham?

Reply:
![image](https://user-images.githubusercontent.com/884550/184018959-543891af-ce1e-4f42-a958-68eef122ee04.jpeg)